pkg/apis/monitoring/v1/thanos_types.go

```diff
@@ -82,6 +82,8 @@ type ThanosRulerSpec struct {
 	Tolerations []v1.Toleration `json:"tolerations,omitempty"`
 	// If specified, the pod's topology spread constraints.
 	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+	// DNS custom configuration to use for a Thanos pod.
+	DNSConfig *v1.PodDNSConfig `json: dnsConfig, omitempty`
```

Comment:
This is incorrect and causing generator failures.

```suggestion
	DNSConfig *v1.PodDNSConfig `json:"dnsConfig,omitempty"`
```